sycl/test/extensions/annotated_arg/annotated_ptr.cpp

```diff
@@ -110,6 +110,9 @@ void TestVectorAddWithAnnotatedMMHosts() {
   // annotated_ptr<int, decltype(properties{dwidth<32>})> arg24(tmp21,
   // properties{dwidth<32>});   // ERR
 
+  // Implicit conversion
+  a1 = raw;
+
```

Comment:
If this doesn't have an expected warning/error associated with it, what is this testing?

I'm just trying to think ahead to what happens if we remove this function in future.  Won't the test start failing?

Reply:
This is testing that the implicit conversion and copy constructor is working. This wasn't the case because:
- the assignment operator was broken
- the copy constructor wasn't const

Reply:
Ah, I see.  I hadn't looked at Part 3.  Thanks.